src/WebJobs.Script/Rpc/LanguageWorkerChannelInitializationService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class LanguageWorkerChannelInitializationService : IHostedService
+    {
+        private readonly IEnumerable<string> _languages = new List<string>()
+        {
+            LanguageWorkerConstants.JavaLanguageWorkerName
+        };
+
+        private readonly IEnumerable<string> _languagesSupportedOnLinux = new List<string>();
+
+        private readonly string _currentLanguageWorkerRuntime;
+
+        private ILanguageWorkerChannelManager _languageWorkerChannelManager;
+
+        public LanguageWorkerChannelInitializationService(ILanguageWorkerChannelManager languageWorkerChannelManager)
+        {
+            _languageWorkerChannelManager = languageWorkerChannelManager ?? throw new ArgumentNullException(nameof(languageWorkerChannelManager));
+            _currentLanguageWorkerRuntime = SystemEnvironment.Instance.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
```

Comment:
Should this cancellation token be flowed into the channel manager?

Reply:
We are not using Cancellation token now as part of initialization. Will add this when we start supporting graceful shutdown for a language worker. Right now, we just shutdown the process.